gguf-py/gguf/utility.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Iterator
+
+if TYPE_CHECKING:
+    from torch import Tensor
+
+
+def fill_templated_filename(filename: str, output_type: str):
+    # Given a file name fill in any type templates e.g. 'some-model-name.{ftype}.gguf'
+    ftype_uppercase: str = output_type.upper()
+    ftype_lowercase: str = output_type.lower()
+    return filename.format(ftype_lowercase,
+                           outtype=ftype_lowercase, ftype=ftype_lowercase,
+                           OUTTYPE=ftype_uppercase, FTYPE=ftype_uppercase)
+
+
+def per_model_weight_count_estimation(tensors: Iterator[tuple[str, Tensor]], expert_count: int) -> int:
```

Comment:
I know you moved this here because it was suggested, but I think it fits better as a method of `Model`, because I feel like counting the parameters is something that should be overridable in the various `Model` subclasses.